### PR TITLE
Fix bugs in manual and auto saving

### PR DESCRIPTION
**Pull request by ChrisP19**
- Made 'saved' text disappear if user takes a new action, e.g. making a new box

- Fixed manual saving issue where boxes not saved disappear after disconnect

- Fixed autosave issue where user could 'unfreeze' screen after disconnect

### CODE REVIEW

File: app/src/js/common/synchronizer.ts [hunk 1]
```diff
@@ -42,6 +42,10 @@ export class Synchronizer {
         self.actionQueue.push(action)
         if (Session.autosave) {
           self.sendActions()
+        } else {
+          if (types.TASK_ACTION_TYPES.includes(action.type)) {
```
Comment (gycn) [inline app/src/js/common/synchronizer.ts]: Could you change this to `else if`?

Comment (ChrisP19) [reply app/src/js/common/synchronizer.ts]: Fixed

File: app/src/js/common/session.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ import { configureStore } from './configure_store'
 import { Track } from './track'
 
 export const enum ConnectionStatus {
-  SAVED, SAVING, RECONNECTING, UNSAVED
+  JUST_SAVED, SAVED, SAVING, RECONNECTING, UNSAVED
```
Comment (gycn) [inline app/src/js/common/session.ts]: Can you rename `JUST_SAVED` to `NOTIFY_SAVED`?

Comment (ChrisP19) [reply app/src/js/common/session.ts]: Changed

**Comment by fyu** [comment]
Flowchart for manual saving by @ChrisP19 

State Diagrams (1)

**Pull request merged**